src/Driver/OCI8/Result.php

```diff
@@ -136,4 +154,46 @@ private function fetchAll(int $mode, int $fetchStructure): array
 
         return $result;
     }
+
+    /**
+     * Screen PHP error messages for any error that the OCI8 extension
+     * doesn't report via oci_error()
+     *
+     * @link http://www.dba-oracle.com/t_error_code_list.htm
+     */
+    public static function handleError(
+        int $errno,
+        string $errstr,
+        string $errfile,
+        int $errline
+    ): bool {
+        if (preg_match_all('/ORA-(\d+)/', $errstr, $matches) === 0) {
+            // If no ORA error codes are present in error message,
+            // skip this error and bubble this error up to
+            // the normal error handler
+            return false;
+        }
+
+        foreach ($matches[1] as $oraErrorCode) {
+            switch ($oraErrorCode) {
+                case '04061':
+                case '04065':
+                case '04068':
+                    throw PHPError::new(
+                        'There was an error before all rows could be fetched.',
+                        $errno,
+                        $errstr,
+                        $errfile,
+                        $errline
+                    );
+                default:
+                    break;
```

Comment:
It looks like the `foreach` loop will either throw an exception or break at the very first iteration. Why do we need `preg_match_all()` then?

If we need to look for any known error, the loop shouldn't break by default. If we want to consider only the first encountered error code, there shouldn't be a loop.

Reply:
Good catch, I added in the break to deal with a code style failure.  Since the error message could contain multiple ORA codes, like the example given, we want to loop over all codes in case the first code doesn't match the block list.  I'll change that to properly continue the loop.